plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt

```diff
@@ -521,68 +586,80 @@ class DefaultPsiToDocumentableTranslator(
+
+            return parseField(
+                psi = psi,
+                getter = getter,
+                setter = setter,
+                inheritedFrom = inheritedFrom
+            )
+        }
 
-        private fun PsiClass.splitFunctionsAndAccessors(): Pair<MutableList<PsiMethod>, MutableMap<PsiField, MutableList<PsiMethod>>> {
-            val fieldNames = fields.associateBy { it.name }
-            val accessors = mutableMapOf<PsiField, MutableList<PsiMethod>>()
-            val regularMethods = mutableListOf<PsiMethod>()
-            methods.forEach { method ->
-                val field = method.getPropertyNameForFunction()?.let { name -> fieldNames[name] }
-                if (field != null) {
-                    accessors.getOrPut(field, ::mutableListOf).add(method)
-                } else {
-                    regularMethods.add(method)
-                }
-            }
-            return regularMethods to accessors
+        private fun parseField(psi: PsiField, accessors: List<PsiMethod>, inheritedFrom: DRI? = null): DProperty {
+            val getter = accessors.firstOrNull { it.isGetterFor(psi) }?.let { parseFunction(it) }
+            val setter = accessors.firstOrNull { it.isSetterFor(psi) }?.let { parseFunction(it) }
+            return parseField(
+                psi = psi,
+                getter = getter,
+                setter = setter,
+                inheritedFrom = inheritedFrom
+            )
         }
 
-        private fun parseField(psi: PsiField, accessors: List<PsiMethod>): DProperty {
+        private fun parseField(psi: PsiField, getter: DFunction?, setter: DFunction?, inheritedFrom: DRI? = null): DProperty {
             val dri = DRI.from(psi)
             return DProperty(
-                dri,
-                psi.name,
-                javadocParser.parseDocumentation(psi).toSourceSetDependent(),
-                null,
-                PsiDocumentableSource(psi).toSourceSetDependent(),
-                psi.getVisibility().toSourceSetDependent(),
-                getBound(psi.type),
-                null,
-                accessors.firstOrNull { it.hasParameters() }?.let { parseFunction(it) },
-                accessors.firstOrNull { it.returnType == psi.type }?.let { parseFunction(it) },
-                psi.getModifier().toSourceSetDependent(),
-                setOf(sourceSetData),
-                emptyList(),
-                false,
-                psi.additionalExtras().let {
+                dri = dri,
+                name = psi.name,
+                documentation = javadocParser.parseDocumentation(psi).toSourceSetDependent(),
+                expectPresentInSet = null,
+                sources = PsiDocumentableSource(psi).toSourceSetDependent(),
+                visibility = psi.getVisibility().toSourceSetDependent(),
+                type = getBound(psi.type),
+                receiver = null,
+                setter = setter,
+                getter = getter,
```

Comment:
Before adding parameter names, these two were mixed, so `setter = getter` and `getter = setter`. Fortunately, the order was mixed in the arguments as well, so all the tests passed.

To avoid such problems in the future reviews and modifications, added parameter names to all documentable creation